source/mainnet/smart-contracts/guides/unit-test-contract.rst

```diff
@@ -482,11 +482,193 @@ account:
         let amount1 = Amount::from_ccd(20);
         claim_eq!(host.get_transfers_to(receiver0), [amount0, amount1]);
 
+.. _writing_property_based_tests:
+
+Writing property-based tests
+============================
+
+The property-based testing technique allows for testing statements about your code that are expected to be true for any input parameters.
```

Comment:
I am not sure this is very meaningful. Quite often one assumes some relationship between the inputs (i.e., ensure some precondition). So this phrasing is a bit misleading in my view.

Reply:
If one has a precondition then the property is an implication. When the precondition is not met, the test is discarded (that is, holds vacuously). 
I can make it more explicit that for any input that possibly satisfies certain validity criteria. Would it be better?

Reply:
I've added an explanation, a general "template" and an example of a property.